apis/github

```diff
@@ -1,19 +1,19 @@
 local JSON = dofile("apis/dkjson")
 
--- Build a github API url, trying to authenticate (not tested)
+local function getAuthHeaders(auth)
```

Comment:
This is no longer needed for DRY but I still think it makes sense to keep it to separate concerns, and lower the complexity of `getAPI`, do you agree?